Sources/NIOCore/ChannelPipeline.swift

```diff
@@ -1082,6 +1082,23 @@ extension ChannelPipeline {
             try self._pipeline.addHandlersSync(handlers, position: position).get()
         }
 
+        /// Remove a `ChannelHandler` from the `ChannelPipeline`.
+        ///
+        /// - parameters:
+        ///     - handler: the `ChannelHandler` to remove.
+        /// - returns: the `EventLoopFuture` which will be notified once the `ChannelHandler` was removed.
+        @preconcurrency
+        public func removeHandler(_ handler: RemovableChannelHandler) -> EventLoopFuture<Void> {
```

Comment:
This is the most relevant change in this PR since it adds a new `removeHandler` method to the `syncOperations`. Importantly this still returns a future since removing is asynchronous but it requires the call to  this method to be from the event loop of the pipeline.